src/main/java/org/jboss/tools/intellij/openshift/utils/odo/OdoProjectDecorator.java

```diff
@@ -1,300 +0,0 @@
- * and is available at http://www.eclipse.org/legal/epl-v20.html
- *
- * Contributors:
- * Red Hat, Inc. - initial API and implementation
- ******************************************************************************/
-package org.jboss.tools.intellij.openshift.utils.odo;
-
-import com.fasterxml.jackson.databind.node.ObjectNode;
-import com.intellij.openapi.util.SystemInfo;
-import com.intellij.openapi.vfs.VfsUtil;
-import com.intellij.openapi.vfs.VirtualFile;
-import org.apache.commons.io.FileUtils;
-import org.apache.commons.lang.StringUtils;
-import org.jboss.tools.intellij.openshift.tree.application.ApplicationsRootNode;
-
-import java.io.File;
-import java.io.IOException;
-import java.nio.file.Files;
-import java.nio.file.attribute.FileAttribute;
-import java.nio.file.attribute.PosixFilePermission;
-import java.nio.file.attribute.PosixFilePermissions;
-import java.util.List;
-import java.util.Map;
-import java.util.Optional;
-import java.util.Set;
-import java.util.function.Consumer;
-
-import static org.jboss.tools.intellij.openshift.Constants.DebugStatus;
-
-public class OdoProjectDecorator implements Odo {
-    private final Odo delegate;
-    private final ApplicationsRootNode root;
-
-    public OdoProjectDecorator(Odo delegate, ApplicationsRootNode root) {
-        this.delegate = delegate;
-        this.root = root;
-    }
-
-    @Override
-    public List<String> getNamespaces() throws IOException {
-        return delegate.getNamespaces();
-    }
-
-    @Override
-    public String getNamespace() throws IOException {
-        return delegate.getNamespace();
-    }
-
-    @Override
-    public void start(String project, String context, String component, ComponentFeature feature,
-                      Consumer<Boolean> callback, Consumer<Boolean> processTerminatedCallback) throws IOException {
-        delegate.start(project, context, component, feature, callback, processTerminatedCallback);
-    }
-
-    @Override
-    public void stop(String project, String context, String component, ComponentFeature feature) throws IOException {
-        delegate.stop(project, context, component, feature);
-    }
-
-    @Override
-    public boolean isStarted(String project, String context, String component, ComponentFeature feature) {
-        return delegate.isStarted(project, context, component, feature);
-    }
-
-    @Override
-    public void describeComponent(String project, String context, String component) throws IOException {
-        delegate.describeComponent(project, context, component);
-    }
-
-    @Override
-    public List<ComponentMetadata> analyze(String path) throws IOException {
-        return delegate.analyze(path);
-    }
-
-    @Override
-    public void createComponent(String project, String componentType, String registryName, String component, String source, String devfile, String starter) throws IOException {
-        if (StringUtils.isNotBlank(starter)) {
```

Comment:
please move this logic in OdoCli or in the calling action

Reply:
I decided to keep `OdoProjectDecorator`. See arguments in comments below.